src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -626,7 +634,7 @@ private Compilation CreateEmptyCompilation()
                 }
             }
 
-            private async Task<(Compilation compilationWithoutGenerators, Compilation? compilationWithGenerators, GeneratorDriver? generatorDriver)> BuildDeclarationCompilationFromInProgressAsync(
+            private async Task<AllSyntaxTreesParsedState> BuildAllSyntaxTreesParsedStateFromInProgressStateAsync(
                 InProgressState state, CancellationToken cancellationToken)
```

Comment:
@jasonmalinowski this was the interesting acse.  where we can prove that we always go from InProgress -> to AllSyntaxTreesParsed -> to FinalState.